### DISCUSSION TITLE
Exit code `1` when running with `--version`

### DISCUSSION THREAD

**Discussion by nhedger** [pacak/bpaf#296] [General]
While publishing a CLI on Homebrew, it was pointed out that the our CLI exits with code `1` when running with only `--version`.

I believe this behavior to be caused by this: https://github.com/pacak/bpaf/blob/8a948b1510a73fd61f6aeb58c427edcfadab5c11/src/info.rs#L285

Is there a reason we don't return exit code `0` in such a case?

Thanks for your insights.

**Comment by pacak** [comment]
```console
$ cargo asm --version
Version: 0.2.20
$ echo $?
0
```

It should produce 0 if application sets version number in some way. It can produce 1 if version is not set and parse fails.

You can specify a version like so: https://docs.rs/bpaf/0.9.5/bpaf/struct.OptionParser.html#method.version

**Reply by nhedger** [reply]
Ah, I must have misunderstood! Your help is much appreciated. I'll try this.

Thanks for a nice lib.
